### ISSUE TITLE
null pre- and postfix on Chatbox client

### ISSUE THREAD

**Issue by bitfl0wer** [aaamoon/copilot-gpt4-service#121]
### Description

I set up this service using docker compose using the following docker-compose.yml

```yml
version: '3'

services:
  copilot-gpt4-service:
    build:
      context: .
      dockerfile: Dockerfile
    environment:
      - HOST=0.0.0.0
    ports:
      - 4780:8080
    restart: unless-stopped
    container_name: copilot-gpt4-service
```

Because I do not want to use Vercel, I decided to use the Chatbox client. This seems to work great, even with GPT4, however, API-request results seem to be pre- and postfixed with `null`, like in the image below.

<img width="1277" alt="Screenshot depicting the Chatbox client and a ChatGPT conversation 'nullChatbot Intronull'." src="https://github.com/aaamoon/copilot-gpt4-service/assets/39242991/f6d67642-5075-4e53-80cc-948730e56aac">

Edit: The following settings are being used in the Chatbox client:

<img width="575" alt="grafik" src="https://github.com/aaamoon/copilot-gpt4-service/assets/39242991/204586c1-1f26-4ad4-89e3-fcbfc79c54af">

**Comment by Geniucker** [comment]
The cause is that the ["choices"]["delta"]["content"] field in the second and penultamate items is null.  
Normally it's handled by chat apps like ChatGPT-Next-Web but it's obvious not here.  
Here's the event stream I got bypackage capturing
```
data: {"object": "chat.completion.chunk", "model": "gpt-3.5-turbo", "choices":[],"created":0,"id":"","prompt_filter_results":[{"content_filter_results":{"hate":{"filtered":false,"severity":"safe"},"self_harm":{"filtered":false,"severity":"safe"},"sexual":{"filtered":false,"severity":"safe"},"violence":{"filtered":false,"severity":"safe"}},"prompt_index":0}]}

data: {"object": "chat.completion.chunk", "model": "gpt-3.5-turbo", "choices":[{"index":0,"content_filter_offsets":{"check_offset":1436,"start_offset":1436,"end_offset":1483},"content_filter_results":{"hate":{"filtered":false,"severity":"safe"},"self_harm":{"filtered":false,"severity":"safe"},"sexual":{"filtered":false,"severity":"safe"},"violence":{"filtered":false,"severity":"safe"}},"delta":{"content":null,"role":"assistant"}}],"created":1704940487,"id":"chatcmpl-8ffC3g4CLFRXUsYiSED6GiKIOx8fp"}

data: {"object": "chat.completion.chunk", "model": "gpt-3.5-turbo", "choices":[{"index":0,"content_filter_offsets":{"check_offset":1436,"start_offset":1436,"end_offset":1483},"content_filter_results":{"hate":{"filtered":false,"severity":"safe"},"self_harm":{"filtered":false,"severity":"safe"},"sexual":{"filtered":false,"severity":"safe"},"violence":{"filtered":false,"severity":"safe"}},"delta":{"content":"非","role":null}}],"created":1704940487,"id":"chatcmpl-8ffC3g4CLFRXUsYiSED6GiKIOx8fp"}

data: {"object": "chat.completion.chunk", "model": "gpt-3.5-turbo", "choices":[{"index":0,"content_filter_offsets":{"check_offset":1436,"start_offset":1436,"end_offset":1483},"content_filter_results":{"hate":{"filtered":false,"severity":"safe"},"self_harm":{"filtered":false,"severity":"safe"},"sexual":{"filtered":false,"severity":"safe"},"violence":{"filtered":false,"severity":"safe"}},"delta":{"content":"常","role":null}}],"created":1704940487,"id":"chatcmpl-8ffC3g4CLFRXUsYiSED6GiKIOx8fp"}

data: {"object": "chat.completion.chunk", "model": "gpt-3.5-turbo", "choices":[{"index":0,"content_filter_offsets":{"check_offset":1436,"start_offset":1436,"end_offset":1483},"content_filter_results":{"hate":{"filtered":false,"severity":"safe"},"self_harm":{"filtered":false,"severity":"safe"},"sexual":{"filtered":false,"severity":"safe"},"violence":{"filtered":false,"severity":"safe"}},"delta":{"content":"感","role":null}}],"created":1704940487,"id":"chatcmpl-8ffC3g4CLFRXUsYiSED6GiKIOx8fp"}

data: {"object": "chat.completion.chunk", "model": "gpt-3.5-turbo", "choices":[{"index":0,"content_filter_offsets":{"check_offset":1436,"start_offset":1436,"end_offset":1483},"content_filter_results":{"hate":{"filtered":false,"severity":"safe"},"self_harm":{"filtered":false,"severity":"safe"},"sexual":{"filtered":false,"severity":"safe"},"violence":{"filtered":false,"severity":"safe"}},"delta":{"content":"谢","role":null}}],"created":1704940487,"id":"chatcmpl-8ffC3g4CLFRXUsYiSED6GiKIOx8fp"}

data: {"object": "chat.completion.chunk", "model": "gpt-3.5-turbo", "choices":[{"index":0,"content_filter_offsets":{"check_offset":1436,"start_offset":1436,"end_offset":1483},"content_filter_results":{"hate":{"filtered":false,"severity":"safe"},"self_harm":{"filtered":false,"severity":"safe"},"sexual":{"filtered":false,"severity":"safe"},"violence":{"filtered":false,"severity":"safe"}},"delta":{"content":"您","role":null}}],"created":1704940487,"id":"chatcmpl-8ffC3g4CLFRXUsYiSED6GiKIOx8fp"}

data: {"object": "chat.completion.chunk", "model": "gpt-3.5-turbo", "choices":[{"index":0,"content_filter_offsets":{"check_offset":1436,"start_offset":1436,"end_offset":1483},"content_filter_results":{"hate":{"filtered":false,"severity":"safe"},"self_harm":{"filtered":false,"severity":"safe"},"sexual":{"filtered":false,"severity":"safe"},"violence":{"filtered":false,"severity":"safe"}},"delta":{"content":"的","role":null}}],"created":1704940487,"id":"chatcmpl-8ffC3g4CLFRXUsYiSED6GiKIOx8fp"}

data: {"object": "chat.completion.chunk", "model": "gpt-3.5-turbo", "choices":[{"index":0,"content_filter_offsets":{"check_offset":1436,"start_offset":1436,"end_offset":1483},"content_filter_results":{"hate":{"filtered":false,"severity":"safe"},"self_harm":{"filtered":false,"severity":"safe"},"sexual":{"filtered":false,"severity":"safe"},"violence":{"filtered":false,"severity":"safe"}},"delta":{"content":"配","role":null}}],"created":1704940487,"id":"chatcmpl-8ffC3g4CLFRXUsYiSED6GiKIOx8fp"}

data: {"object": "chat.completion.chunk", "model": "gpt-3.5-turbo", "choices":[{"index":0,"content_filter_offsets":{"check_offset":1436,"start_offset":1436,"end_offset":1483},"content_filter_results":{"hate":{"filtered":false,"severity":"safe"},"self_harm":{"filtered":false,"severity":"safe"},"sexual":{"filtered":false,"severity":"safe"},"violence":{"filtered":false,"severity":"safe"}},"delta":{"content":"合","role":null}}],"created":1704940487,"id":"chatcmpl-8ffC3g4CLFRXUsYiSED6GiKIOx8fp"}

data: {"object": "chat.completion.chunk", "model": "gpt-3.5-turbo", "choices":[{"index":0,"content_filter_offsets":{"check_offset":1436,"start_offset":1436,"end_offset":1483},"content_filter_results":{"hate":{"filtered":false,"severity":"safe"},"self_harm":{"filtered":false,"severity":"safe"},"sexual":{"filtered":false,"severity":"safe"},"violence":{"filtered":false,"severity":"safe"}},"delta":{"content":"！","role":null}}],"created":1704940487,"id":"chatcmpl-8ffC3g4CLFRXUsYiSED6GiKIOx8fp"}

data: {"object": "chat.completion.chunk", "model": "gpt-3.5-turbo", "choices":[{"index":0,"content_filter_offsets":{"check_offset":1436,"start_offset":1436,"end_offset":1483},"content_filter_results":{"hate":{"filtered":false,"severity":"safe"},"self_harm":{"filtered":false,"severity":"safe"},"sexual":{"filtered":false,"severity":"safe"},"violence":{"filtered":false,"severity":"safe"}},"delta":{"content":"如果","role":null}}],"created":1704940487,"id":"chatcmpl-8ffC3g4CLFRXUsYiSED6GiKIOx8fp"}

data: {"object": "chat.completion.chunk", "model": "gpt-3.5-turbo", "choices":[{"index":0,"content_filter_offsets":{"check_offset":1436,"start_offset":1436,"end_offset":1483},"content_filter_results":{"hate":{"filtered":false,"severity":"safe"},"self_harm":{"filtered":false,"severity":"safe"},"sexual":{"filtered":false,"severity":"safe"},"violence":{"filtered":false,"severity":"safe"}},"delta":{"content":"您","role":null}}],"created":1704940487,"id":"chatcmpl-8ffC3g4CLFRXUsYiSED6GiKIOx8fp"}

data: {"object": "chat.completion.chunk", "model": "gpt-3.5-turbo", "choices":[{"index":0,"content_filter_offsets":{"check_offset":1436,"start_offset":1436,"end_offset":1483},"content_filter_results":{"hate":{"filtered":false,"severity":"safe"},"self_harm":{"filtered":false,"severity":"safe"},"sexual":{"filtered":false,"severity":"safe"},"violence":{"filtered":false,"severity":"safe"}},"delta":{"content":"需要","role":null}}],"created":1704940487,"id":"chatcmpl-8ffC3g4CLFRXUsYiSED6GiKIOx8fp"}

data: {"object": "chat.completion.chunk", "model": "gpt-3.5-turbo", "choices":[{"index":0,"content_filter_offsets":{"check_offset":1436,"start_offset":1436,"end_offset":1483},"content_filter_results":{"hate":{"filtered":false,"severity":"safe"},"self_harm":{"filtered":false,"severity":"safe"},"sexual":{"filtered":false,"severity":"safe"},"violence":{"filtered":false,"severity":"safe"}},"delta":{"content":"任","role":null}}],"created":1704940487,"id":"chatcmpl-8ffC3g4CLFRXUsYiSED6GiKIOx8fp"}

data: {"object": "chat.completion.chunk", "model": "gpt-3.5-turbo", "choices":[{"index":0,"content_filter_offsets":{"check_offset":1436,"start_offset":1436,"end_offset":1483},"content_filter_results":{"hate":{"filtered":false,"severity":"safe"},"self_harm":{"filtered":false,"severity":"safe"},"sexual":{"filtered":false,"severity":"safe"},"violence":{"filtered":false,"severity":"safe"}},"delta":{"content":"何","role":null}}],"created":1704940487,"id":"chatcmpl-8ffC3g4CLFRXUsYiSED6GiKIOx8fp"}

data: {"object": "chat.completion.chunk", "model": "gpt-3.5-turbo", "choices":[{"index":0,"content_filter_offsets":{"check_offset":1436,"start_offset":1436,"end_offset":1483},"content_filter_results":{"hate":{"filtered":false,"severity":"safe"},"self_harm":{"filtered":false,"severity":"safe"},"sexual":{"filtered":false,"severity":"safe"},"violence":{"filtered":false,"severity":"safe"}},"delta":{"content":"帮","role":null}}],"created":1704940487,"id":"chatcmpl-8ffC3g4CLFRXUsYiSED6GiKIOx8fp"}

data: {"object": "chat.completion.chunk", "model": "gpt-3.5-turbo", "choices":[{"index":0,"content_filter_offsets":{"check_offset":1436,"start_offset":1436,"end_offset":1483},"content_filter_results":{"hate":{"filtered":false,"severity":"safe"},"self_harm":{"filtered":false,"severity":"safe"},"sexual":{"filtered":false,"severity":"safe"},"violence":{"filtered":false,"severity":"safe"}},"delta":{"content":"助","role":null}}],"created":1704940487,"id":"chatcmpl-8ffC3g4CLFRXUsYiSED6GiKIOx8fp"}

data: {"object": "chat.completion.chunk", "model": "gpt-3.5-turbo", "choices":[{"index":0,"content_filter_offsets":{"check_offset":1436,"start_offset":1436,"end_offset":1483},"content_filter_results":{"hate":{"filtered":false,"severity":"safe"},"self_harm":{"filtered":false,"severity":"safe"},"sexual":{"filtered":false,"severity":"safe"},"violence":{"filtered":false,"severity":"safe"}},"delta":{"content":"或","role":null}}],"created":1704940487,"id":"chatcmpl-8ffC3g4CLFRXUsYiSED6GiKIOx8fp"}

data: {"object": "chat.completion.chunk", "model": "gpt-3.5-turbo", "choices":[{"index":0,"content_filter_offsets":{"check_offset":1436,"start_offset":1436,"end_offset":1483},"content_filter_results":{"hate":{"filtered":false,"severity":"safe"},"self_harm":{"filtered":false,"severity":"safe"},"sexual":{"filtered":false,"severity":"safe"},"violence":{"filtered":false,"severity":"safe"}},"delta":{"content":"有","role":null}}],"created":1704940487,"id":"chatcmpl-8ffC3g4CLFRXUsYiSED6GiKIOx8fp"}

data: {"object": "chat.completion.chunk", "model": "gpt-3.5-turbo", "choices":[{"index":0,"content_filter_offsets":{"check_offset":1436,"start_offset":1436,"end_offset":1483},"content_filter_results":{"hate":{"filtered":false,"severity":"safe"},"self_harm":{"filtered":false,"severity":"safe"},"sexual":{"filtered":false,"severity":"safe"},"violence":{"filtered":false,"severity":"safe"}},"delta":{"content":"任","role":null}}],"created":1704940487,"id":"chatcmpl-8ffC3g4CLFRXUsYiSED6GiKIOx8fp"}

data: {"object": "chat.completion.chunk", "model": "gpt-3.5-turbo", "choices":[{"index":0,"content_filter_offsets":{"check_offset":1436,"start_offset":1436,"end_offset":1483},"content_filter_results":{"hate":{"filtered":false,"severity":"safe"},"self_harm":{"filtered":false,"severity":"safe"},"sexual":{"filtered":false,"severity":"safe"},"violence":{"filtered":false,"severity":"safe"}},"delta":{"content":"何","role":null}}],"created":1704940487,"id":"chatcmpl-8ffC3g4CLFRXUsYiSED6GiKIOx8fp"}

data: {"object": "chat.completion.chunk", "model": "gpt-3.5-turbo", "choices":[{"index":0,"content_filter_offsets":{"check_offset":1436,"start_offset":1436,"end_offset":1483},"content_filter_results":{"hate":{"filtered":false,"severity":"safe"},"self_harm":{"filtered":false,"severity":"safe"},"sexual":{"filtered":false,"severity":"safe"},"violence":{"filtered":false,"severity":"safe"}},"delta":{"content":"问题","role":null}}],"created":1704940487,"id":"chatcmpl-8ffC3g4CLFRXUsYiSED6GiKIOx8fp"}

data: {"object": "chat.completion.chunk", "model": "gpt-3.5-turbo", "choices":[{"index":0,"content_filter_offsets":{"check_offset":1436,"start_offset":1436,"end_offset":1483},"content_filter_results":{"hate":{"filtered":false,"severity":"safe"},"self_harm":{"filtered":false,"severity":"safe"},"sexual":{"filtered":false,"severity":"safe"},"violence":{"filtered":false,"severity":"safe"}},"delta":{"content":"，请","role":null}}],"created":1704940487,"id":"chatcmpl-8ffC3g4CLFRXUsYiSED6GiKIOx8fp"}

data: {"object": "chat.completion.chunk", "model": "gpt-3.5-turbo", "choices":[{"index":0,"content_filter_offsets":{"check_offset":1436,"start_offset":1436,"end_offset":1483},"content_filter_results":{"hate":{"filtered":false,"severity":"safe"},"self_harm":{"filtered":false,"severity":"safe"},"sexual":{"filtered":false,"severity":"safe"},"violence":{"filtered":false,"severity":"safe"}},"delta":{"content":"随","role":null}}],"created":1704940487,"id":"chatcmpl-8ffC3g4CLFRXUsYiSED6GiKIOx8fp"}

data: {"object": "chat.completion.chunk", "model": "gpt-3.5-turbo", "choices":[{"index":0,"content_filter_offsets":{"check_offset":1436,"start_offset":1436,"end_offset":1483},"content_filter_results":{"hate":{"filtered":false,"severity":"safe"},"self_harm":{"filtered":false,"severity":"safe"},"sexual":{"filtered":false,"severity":"safe"},"violence":{"filtered":false,"severity":"safe"}},"delta":{"content":"时","role":null}}],"created":1704940487,"id":"chatcmpl-8ffC3g4CLFRXUsYiSED6GiKIOx8fp"}

data: {"object": "chat.completion.chunk", "model": "gpt-3.5-turbo", "choices":[{"index":0,"content_filter_offsets":{"check_offset":1436,"start_offset":1436,"end_offset":1483},"content_filter_results":{"hate":{"filtered":false,"severity":"safe"},"self_harm":{"filtered":false,"severity":"safe"},"sexual":{"filtered":false,"severity":"safe"},"violence":{"filtered":false,"severity":"safe"}},"delta":{"content":"告","role":null}}],"created":1704940487,"id":"chatcmpl-8ffC3g4CLFRXUsYiSED6GiKIOx8fp"}

data: {"object": "chat.completion.chunk", "model": "gpt-3.5-turbo", "choices":[{"index":0,"content_filter_offsets":{"check_offset":1436,"start_offset":1436,"end_offset":1483},"content_filter_results":{"hate":{"filtered":false,"severity":"safe"},"self_harm":{"filtered":false,"severity":"safe"},"sexual":{"filtered":false,"severity":"safe"},"violence":{"filtered":false,"severity":"safe"}},"delta":{"content":"诉","role":null}}],"created":1704940487,"id":"chatcmpl-8ffC3g4CLFRXUsYiSED6GiKIOx8fp"}

data: {"object": "chat.completion.chunk", "model": "gpt-3.5-turbo", "choices":[{"index":0,"content_filter_offsets":{"check_offset":1436,"start_offset":1436,"end_offset":1483},"content_filter_results":{"hate":{"filtered":false,"severity":"safe"},"self_harm":{"filtered":false,"severity":"safe"},"sexual":{"filtered":false,"severity":"safe"},"violence":{"filtered":false,"severity":"safe"}},"delta":{"content":"我","role":null}}],"created":1704940487,"id":"chatcmpl-8ffC3g4CLFRXUsYiSED6GiKIOx8fp"}

data: {"object": "chat.completion.chunk", "model": "gpt-3.5-turbo", "choices":[{"index":0,"content_filter_offsets":{"check_offset":1436,"start_offset":1436,"end_offset":1483},"content_filter_results":{"hate":{"filtered":false,"severity":"safe"},"self_harm":{"filtered":false,"severity":"safe"},"sexual":{"filtered":false,"severity":"safe"},"violence":{"filtered":false,"severity":"safe"}},"delta":{"content":"。","role":null}}],"created":1704940487,"id":"chatcmpl-8ffC3g4CLFRXUsYiSED6GiKIOx8fp"}

data: {"object": "chat.completion.chunk", "model": "gpt-3.5-turbo", "choices":[{"index":0,"content_filter_offsets":{"check_offset":1436,"start_offset":1436,"end_offset":1483},"content_filter_results":{"hate":{"filtered":false,"severity":"safe"},"self_harm":{"filtered":false,"severity":"safe"},"sexual":{"filtered":false,"severity":"safe"},"violence":{"filtered":false,"severity":"safe"}},"delta":{"content":"我","role":null}}],"created":1704940487,"id":"chatcmpl-8ffC3g4CLFRXUsYiSED6GiKIOx8fp"}

data: {"object": "chat.completion.chunk", "model": "gpt-3.5-turbo", "choices":[{"index":0,"content_filter_offsets":{"check_offset":1436,"start_offset":1436,"end_offset":1483},"content_filter_results":{"hate":{"filtered":false,"severity":"safe"},"self_harm":{"filtered":false,"severity":"safe"},"sexual":{"filtered":false,"severity":"safe"},"violence":{"filtered":false,"severity":"safe"}},"delta":{"content":"将","role":null}}],"created":1704940487,"id":"chatcmpl-8ffC3g4CLFRXUsYiSED6GiKIOx8fp"}

data: {"object": "chat.completion.chunk", "model": "gpt-3.5-turbo", "choices":[{"index":0,"content_filter_offsets":{"check_offset":1436,"start_offset":1436,"end_offset":1483},"content_filter_results":{"hate":{"filtered":false,"severity":"safe"},"self_harm":{"filtered":false,"severity":"safe"},"sexual":{"filtered":false,"severity":"safe"},"violence":{"filtered":false,"severity":"safe"}},"delta":{"content":"尽","role":null}}],"created":1704940487,"id":"chatcmpl-8ffC3g4CLFRXUsYiSED6GiKIOx8fp"}

data: {"object": "chat.completion.chunk", "model": "gpt-3.5-turbo", "choices":[{"index":0,"content_filter_offsets":{"check_offset":1436,"start_offset":1436,"end_offset":1483},"content_filter_results":{"hate":{"filtered":false,"severity":"safe"},"self_harm":{"filtered":false,"severity":"safe"},"sexual":{"filtered":false,"severity":"safe"},"violence":{"filtered":false,"severity":"safe"}},"delta":{"content":"力","role":null}}],"created":1704940487,"id":"chatcmpl-8ffC3g4CLFRXUsYiSED6GiKIOx8fp"}

data: {"object": "chat.completion.chunk", "model": "gpt-3.5-turbo", "choices":[{"index":0,"content_filter_offsets":{"check_offset":1436,"start_offset":1436,"end_offset":1483},"content_filter_results":{"hate":{"filtered":false,"severity":"safe"},"self_harm":{"filtered":false,"severity":"safe"},"sexual":{"filtered":false,"severity":"safe"},"violence":{"filtered":false,"severity":"safe"}},"delta":{"content":"提","role":null}}],"created":1704940487,"id":"chatcmpl-8ffC3g4CLFRXUsYiSED6GiKIOx8fp"}

data: {"object": "chat.completion.chunk", "model": "gpt-3.5-turbo", "choices":[{"index":0,"content_filter_offsets":{"check_offset":1436,"start_offset":1436,"end_offset":1483},"content_filter_results":{"hate":{"filtered":false,"severity":"safe"},"self_harm":{"filtered":false,"severity":"safe"},"sexual":{"filtered":false,"severity":"safe"},"violence":{"filtered":false,"severity":"safe"}},"delta":{"content":"供","role":null}}],"created":1704940487,"id":"chatcmpl-8ffC3g4CLFRXUsYiSED6GiKIOx8fp"}

data: {"object": "chat.completion.chunk", "model": "gpt-3.5-turbo", "choices":[{"index":0,"content_filter_offsets":{"check_offset":1436,"start_offset":1436,"end_offset":1483},"content_filter_results":{"hate":{"filtered":false,"severity":"safe"},"self_harm":{"filtered":false,"severity":"safe"},"sexual":{"filtered":false,"severity":"safe"},"violence":{"filtered":false,"severity":"safe"}},"delta":{"content":"所","role":null}}],"created":1704940487,"id":"chatcmpl-8ffC3g4CLFRXUsYiSED6GiKIOx8fp"}

data: {"object": "chat.completion.chunk", "model": "gpt-3.5-turbo", "choices":[{"index":0,"content_filter_offsets":{"check_offset":1436,"start_offset":1436,"end_offset":1483},"content_filter_results":{"hate":{"filtered":false,"severity":"safe"},"self_harm":{"filtered":false,"severity":"safe"},"sexual":{"filtered":false,"severity":"safe"},"violence":{"filtered":false,"severity":"safe"}},"delta":{"content":"需","role":null}}],"created":1704940487,"id":"chatcmpl-8ffC3g4CLFRXUsYiSED6GiKIOx8fp"}

data: {"object": "chat.completion.chunk", "model": "gpt-3.5-turbo", "choices":[{"index":0,"content_filter_offsets":{"check_offset":1436,"start_offset":1436,"end_offset":1483},"content_filter_results":{"hate":{"filtered":false,"severity":"safe"},"self_harm":{"filtered":false,"severity":"safe"},"sexual":{"filtered":false,"severity":"safe"},"violence":{"filtered":false,"severity":"safe"}},"delta":{"content":"的","role":null}}],"created":1704940487,"id":"chatcmpl-8ffC3g4CLFRXUsYiSED6GiKIOx8fp"}

data: {"object": "chat.completion.chunk", "model": "gpt-3.5-turbo", "choices":[{"index":0,"content_filter_offsets":{"check_offset":1436,"start_offset":1436,"end_offset":1483},"content_filter_results":{"hate":{"filtered":false,"severity":"safe"},"self_harm":{"filtered":false,"severity":"safe"},"sexual":{"filtered":false,"severity":"safe"},"violence":{"filtered":false,"severity":"safe"}},"delta":{"content":"支","role":null}}],"created":1704940487,"id":"chatcmpl-8ffC3g4CLFRXUsYiSED6GiKIOx8fp"}

data: {"object": "chat.completion.chunk", "model": "gpt-3.5-turbo", "choices":[{"index":0,"content_filter_offsets":{"check_offset":1436,"start_offset":1436,"end_offset":1483},"content_filter_results":{"hate":{"filtered":false,"severity":"safe"},"self_harm":{"filtered":false,"severity":"safe"},"sexual":{"filtered":false,"severity":"safe"},"violence":{"filtered":false,"severity":"safe"}},"delta":{"content":"持","role":null}}],"created":1704940487,"id":"chatcmpl-8ffC3g4CLFRXUsYiSED6GiKIOx8fp"}

data: {"object": "chat.completion.chunk", "model": "gpt-3.5-turbo", "choices":[{"index":0,"content_filter_offsets":{"check_offset":1436,"start_offset":1436,"end_offset":1483},"content_filter_results":{"hate":{"filtered":false,"severity":"safe"},"self_harm":{"filtered":false,"severity":"safe"},"sexual":{"filtered":false,"severity":"safe"},"violence":{"filtered":false,"severity":"safe"}},"delta":{"content":"和","role":null}}],"created":1704940487,"id":"chatcmpl-8ffC3g4CLFRXUsYiSED6GiKIOx8fp"}

data: {"object": "chat.completion.chunk", "model": "gpt-3.5-turbo", "choices":[{"index":0,"content_filter_offsets":{"check_offset":1436,"start_offset":1436,"end_offset":1483},"content_filter_results":{"hate":{"filtered":false,"severity":"safe"},"self_harm":{"filtered":false,"severity":"safe"},"sexual":{"filtered":false,"severity":"safe"},"violence":{"filtered":false,"severity":"safe"}},"delta":{"content":"解","role":null}}],"created":1704940487,"id":"chatcmpl-8ffC3g4CLFRXUsYiSED6GiKIOx8fp"}

data: {"object": "chat.completion.chunk", "model": "gpt-3.5-turbo", "choices":[{"index":0,"content_filter_offsets":{"check_offset":1436,"start_offset":1436,"end_offset":1483},"content_filter_results":{"hate":{"filtered":false,"severity":"safe"},"self_harm":{"filtered":false,"severity":"safe"},"sexual":{"filtered":false,"severity":"safe"},"violence":{"filtered":false,"severity":"safe"}},"delta":{"content":"答","role":null}}],"created":1704940487,"id":"chatcmpl-8ffC3g4CLFRXUsYiSED6GiKIOx8fp"}

data: {"object": "chat.completion.chunk", "model": "gpt-3.5-turbo", "choices":[{"index":0,"content_filter_offsets":{"check_offset":1436,"start_offset":1436,"end_offset":1483},"content_filter_results":{"hate":{"filtered":false,"severity":"safe"},"self_harm":{"filtered":false,"severity":"safe"},"sexual":{"filtered":false,"severity":"safe"},"violence":{"filtered":false,"severity":"safe"}},"delta":{"content":"。","role":null}}],"created":1704940487,"id":"chatcmpl-8ffC3g4CLFRXUsYiSED6GiKIOx8fp"}

data: {"object": "chat.completion.chunk", "model": "gpt-3.5-turbo", "choices":[{"finish_reason":"stop","index":0,"content_filter_offsets":{"check_offset":1436,"start_offset":1436,"end_offset":1483},"content_filter_results":{"hate":{"filtered":false,"severity":"safe"},"self_harm":{"filtered":false,"severity":"safe"},"sexual":{"filtered":false,"severity":"safe"},"violence":{"filtered":false,"severity":"safe"}},"delta":{"content":null,"role":null}}],"created":1704940487,"id":"chatcmpl-8ffC3g4CLFRXUsYiSED6GiKIOx8fp"}

data: [DONE]
```

**Comment by Geniucker** [comment]
Let's check the resp of OpenAI API
```
data: {"id":"chatcmpl-8ffUrdr1pDgEnYmODt18HzQNwgKY8","object":"chat.completion.chunk","created":1704941653,"model":"gpt-3.5-turbo-0613","system_fingerprint":null,"choices":[{"index":0,"delta":{"role":"assistant","content":""},"logprobs":null,"finish_reason":null}]}

data: {"id":"chatcmpl-8ffUrdr1pDgEnYmODt18HzQNwgKY8","object":"chat.completion.chunk","created":1704941653,"model":"gpt-3.5-turbo-0613","system_fingerprint":null,"choices":[{"index":0,"delta":{"content":"Hello"},"logprobs":null,"finish_reason":null}]}

data: {"id":"chatcmpl-8ffUrdr1pDgEnYmODt18HzQNwgKY8","object":"chat.completion.chunk","created":1704941653,"model":"gpt-3.5-turbo-0613","system_fingerprint":null,"choices":[{"index":0,"delta":{"content":"!"},"logprobs":null,"finish_reason":null}]}

data: {"id":"chatcmpl-8ffUrdr1pDgEnYmODt18HzQNwgKY8","object":"chat.completion.chunk","created":1704941653,"model":"gpt-3.5-turbo-0613","system_fingerprint":null,"choices":[{"index":0,"delta":{"content":" How"},"logprobs":null,"finish_reason":null}]}

data: {"id":"chatcmpl-8ffUrdr1pDgEnYmODt18HzQNwgKY8","object":"chat.completion.chunk","created":1704941653,"model":"gpt-3.5-turbo-0613","system_fingerprint":null,"choices":[{"index":0,"delta":{"content":" can"},"logprobs":null,"finish_reason":null}]}

data: {"id":"chatcmpl-8ffUrdr1pDgEnYmODt18HzQNwgKY8","object":"chat.completion.chunk","created":1704941653,"model":"gpt-3.5-turbo-0613","system_fingerprint":null,"choices":[{"index":0,"delta":{"content":" I"},"logprobs":null,"finish_reason":null}]}

data: {"id":"chatcmpl-8ffUrdr1pDgEnYmODt18HzQNwgKY8","object":"chat.completion.chunk","created":1704941653,"model":"gpt-3.5-turbo-0613","system_fingerprint":null,"choices":[{"index":0,"delta":{"content":" assist"},"logprobs":null,"finish_reason":null}]}

data: {"id":"chatcmpl-8ffUrdr1pDgEnYmODt18HzQNwgKY8","object":"chat.completion.chunk","created":1704941653,"model":"gpt-3.5-turbo-0613","system_fingerprint":null,"choices":[{"index":0,"delta":{"content":" you"},"logprobs":null,"finish_reason":null}]}

data: {"id":"chatcmpl-8ffUrdr1pDgEnYmODt18HzQNwgKY8","object":"chat.completion.chunk","created":1704941653,"model":"gpt-3.5-turbo-0613","system_fingerprint":null,"choices":[{"index":0,"delta":{"content":" today"},"logprobs":null,"finish_reason":null}]}

data: {"id":"chatcmpl-8ffUrdr1pDgEnYmODt18HzQNwgKY8","object":"chat.completion.chunk","created":1704941653,"model":"gpt-3.5-turbo-0613","system_fingerprint":null,"choices":[{"index":0,"delta":{"content":"?"},"logprobs":null,"finish_reason":null}]}

data: {"id":"chatcmpl-8ffUrdr1pDgEnYmODt18HzQNwgKY8","object":"chat.completion.chunk","created":1704941653,"model":"gpt-3.5-turbo-0613","system_fingerprint":null,"choices":[{"index":0,"delta":{},"logprobs":null,"finish_reason":"stop"}]}

data: [DONE]
```

**Comment by bitfl0wer** [comment]
@aaamoon if i may ask, why is this closed as completed if the pr #132 has not yet been merged?

**Comment by aaamoon** [comment]
@bitfl0wer Sorry, I forgot, now it has been merged.
